samcli/commands/init/override_init_flows.py

```diff
@@ -0,0 +1,171 @@
+"""
+Module containing interfaces and implementations for override workflows
+"""
+from abc import ABC, abstractmethod
+from typing import Dict, Optional, Type
+
+import click
+
+from samcli.commands._utils.options import generate_next_command_recommendation
+from samcli.commands.exceptions import InvalidInitOptionException
+from samcli.commands.init.init_flow_helpers import generate_summary_message
+from samcli.commands.init.init_generator import do_generate
+from samcli.commands.init.init_templates import InitTemplates
+from samcli.lib.utils.packagetype import ZIP
+
+NOT_APPLICABLE = "N/A"
+APPSYNC_OVERRIDE_USE_CASE = "GraphQLApi Hello World Example"
+
+
+class OverrideFlow(ABC):
+    """
+    Interface for overriding the default init workflow for app templates that don't conform to the standard flow
+    e.g if an app template doesn't include a Lambda function,
+    many of the questions asked in the default flow are not relevant and can be skipped
+    """
+
+    def __init__(
+        self,
+        templates: InitTemplates,
+        use_case: str,
+        preprocessed_options: dict,
+        name: Optional[str],
+        output_dir: Optional[str],
+    ):
+        self._name = name
+        self._templates = templates
+        self._use_case = use_case
+        self._preprocessed_options = preprocessed_options
+        self._output_dir = output_dir
+
+    @abstractmethod
+    def generate_project(self):
```

Comment:
Is override a short-circuit process? is it able to adapt to adding more questions/prompts if need be?

Reply:
It's a short circuit process. My thinking was that any additional questions beyond the initial use case would be handled by the overriding flow. Happy to hear other ideas for the approach